### PR TITLE
Feat/potions page

### PR DESCRIPTION
**Pull request by eduardconstantin**
- [x] I have read and agree to the [Code of Conduct](https://github.com/danielschuster-muc/potter-db/blob/master/CODE_OF_CONDUCT.md), and have followed the [Contributing guidelines](https://github.com/danielschuster-muc/potter-db/blob/master/CONTRIBUTING.md).

## Pull Request Type

- [x] New feature (either for website or api)

## Summary

- Create pages for potions list and detailed potion information
- Create function to set card border color based on potion difficulty
- Add potions link to footer

### CODE REVIEW

File: frontend/src/components/layout/Footer.js [hunk 1]
```diff
@@ -1,93 +1,87 @@
...
+					link: '/characters',
+				},
+				{
+					name: 'Potions',
+					link: '/potions',
+				},
+				{
+					name: 'Movies',
+					link: '/movies',
+				},
```
Comment (danielschuster-muc) [inline frontend/src/components/layout/Footer.js:43]: Could you please sort those alphabetically? First Characters, then Movies and finally Potions?

File: frontend/src/components/pages/potions/PotionListItem.js [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+						priority
+					/>
+				</CardMedia>
+				<CardHeader title={name} />
+				<CardContent>
+					<ListItemBox value={difficulty || 'Unknown'} icon={faStar} />
+					<ListItemBox value={inventors || 'Unknown'} icon={faPerson} />
+					<ListItemBox value={side_effects || 'Unknown'} icon={faCross} />
+					<ListItemBox value={effect || 'Unknown'} icon={faEdit} />
+					<ListItemBox value={characteristics || 'Unknown'} icon={faPlus} />
```
Comment (danielschuster-muc) [inline frontend/src/components/pages/potions/PotionListItem.js:45]: Can you remove the 'Unknown' from all of the ListItemBoxes? It should only render those if there's a value for each of them, which is done in the ListItemBox itself, so no need to add the 'Unknown'.

File: frontend/src/components/pages/potions/[slug]/PotionDetails.js [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+		ingredients,
+		manufacturers,
+		wiki,
+	} = attributes;
+
+	const accordions = [
+		{
+			name: 'Ingredients',
+			value: ingredients?.split(', '),
+		},
```
Comment (danielschuster-muc) [inline frontend/src/components/pages/potions/[slug]/PotionDetails.js:27]: Maybe we can have 2 or 3 more fields for the accordions, e.g manufacturers, inventors, characteristics etc. Else the accordions is really empty. 
And less things in the informationTable below?

File: frontend/src/lib/utils.js [hunk 4]
```diff
@@ -1,33 +1,52 @@
...
+		case 'Moderate to Advanced' || 'Moderate/Advanced':
+			return '#dbae3b';
+		case 'Advanced':
+			return '#9e1919';
+		case 'One of a kind':
+			return '#21a376';
+		default:
+			return 'transparent';
+	}
 };
```
Comment (danielschuster-muc) [inline frontend/src/lib/utils.js:33]: Setting the potion color by difficulty is a good idea! 
But I am not quite sure about the color palette. It looks like [this ](https://coolors.co/ffffff-bebebe-bdae2d-dbae3b-9e1919-21a376 ) basically, doesn't it? I would maybe prefer a palette from light green (beginner) to dark red (advanced). 
And I would use `#includes(level)` as it there are like 12 different variants of difficulty levels (advanced, advanced to beginner, beginner, beginner to advanced, beginner to moderate, beginner to ordinary wizarding level, moderate to advanced, moderate to ordinary wizarding level, moderate/advanced, one of a kind and ordinary wizarding level). 
That would maybe cover all variants.

File: frontend/src/pages/potions/[slug]/index.js [hunk 5]
```diff
@@ -0,0 +1,77 @@
...
+		'baneberry-potion',
+		'baruffio-s-brain-elixir',
+		'beautification-potion',
+		'befuddlement-draught',
+		'blood-replenishing-potion',
+		'bloodroot-poison',
+		'bruise-removal-paste',
+		'bulgeye-potion',
+		'bundimun-pomade',
+	];
```
Comment (danielschuster-muc) [inline frontend/src/pages/potions/[slug]/index.js:39]: That's not really needed here. I only used in the characters page to preload famous characters and to let search engines get those famous characters. Something like in the [movies page](https://github.com/danielschuster-muc/potter-db/blob/master/frontend/src/pages/movies/%5Bslug%5D/index.js#L16) is probably sufficient.

**Review by danielschuster-muc — CHANGES_REQUESTED** [review]
Thank you for your PR.
In general it looks pretty good 👍, just some small changes. 
One big issue however is that you use 4 or more tab spaces, I use just 2 space for the tabs. And you should change the quotes back to double quotes. Could you please fix those 2 styling issues?

**Comment by eduardconstantin** [comment]
@danielschuster-muc sure, I'll change the format

File: frontend/src/lib/utils.js (hunk 4 above)
Comment (danielschuster-muc) [reply frontend/src/lib/utils.js:33]: What I meant with that is using a the [`include`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/includes) method to check if the difficulty contains one of the difficulty levels. 

like: 
``` 
if difficulty.includes("beginner") {
  return "#fff"
} else difficulty.includes("advanced") {
 return "#bebebe"
} 
...
```
Maybe we can also combine it with a regex or switch, but that'S how I would do it basically.

**Comment by danielschuster-muc** [comment]
> @danielschuster-muc sure, I'll change the format. I don't know why it appears with 4 tabs, in the editor I have 2 spaces. image

Hm, that's very weird. To you have any form of auto format? Maybe that make a problem? Or maybe this can automatically format it to the correct tab spacing?

**Comment by eduardconstantin** [comment]
Could you check if the spacing is ok now?

File: frontend/src/lib/utils.js (hunk 4 above)
Comment (eduardconstantin) [reply frontend/src/lib/utils.js]: I see now, I'll try something

**Comment by danielschuster-muc** [comment]
> Could you check if the spacing is ok now?

The spacing still seems to be wrong. 
Maybe one of those help: https://stackoverflow.com/questions/36251820/visual-studio-code-format-is-not-using-indent-settings

File: frontend/src/lib/utils.js (hunk 4 above)
Comment (eduardconstantin) [reply frontend/src/lib/utils.js]: I put this color palette: https://coolors.co/8ee485-7dc44f-45a85e-d8c624-db6e3b-801c1c

**Comment by eduardconstantin** [comment]
This is how it looks in the editor 
image

File: frontend/src/lib/utils.js (hunk 4 above)
Comment (danielschuster-muc) [reply frontend/src/lib/utils.js]: That's a nice palette!

**Comment by danielschuster-muc** [comment]
Very strange.

**Comment by eduardconstantin** [comment]
ok, I think I found the issue

**Comment by danielschuster-muc** [comment]
> ok, I think I found the issue

That would be great!

**Review by danielschuster-muc — CHANGES_REQUESTED** [review]
Everything else looks great, thank you!

File: frontend/src/lib/utils.js [hunk 6]
```diff
@@ -13,6 +13,21 @@ export const getHouseColor = (house) => {
   }
 };
 
+export const getPotionDifficultyColor = (difficulty) => {
+  switch (true) {
+    case /begginer/i.test(difficulty):
+      return "#7DC44F";
+    case /ordinary/i.test(difficulty):
+      return "#45A85E";
```
Comment (danielschuster-muc) [inline frontend/src/lib/utils.js:21]: an can you replace this color with a yellow one?

**Review by danielschuster-muc — APPROVED** [review]
That's it! Thanks a lot for your contribution, again.

**Comment by eduardconstantin** [comment]
Thank you!

**Pull request merged**